### PR TITLE
Fix caching mechanism inside CSSModules.loadPath

### LINKED ISSUE #63: Files processed more than once
I'm noticing that some of our CSS files are being run through PostCSS more than once.  I'm not sure if it's an issue at this level or at the `broccoli-css-modules` level, but a particular plugin that warns about missing retina images fires multiple times for this one file:

screen shot 2017-01-20 at 5 52 49 pm

If it weren't for the noisy warning, I wouldn't have noticed.

I dug in a bit to see if the issue was with the `postcss` plugin (it's one that I wrote) and as far as I can tell, it's not an issue with the plugin; a `debugger` set to break when that particular file is first loaded up catches 5 separate times, indicating to me that `postcss` is processing the file 5 times.

What's interesting is that the same plugin warns for other files, but only once. I have no great way of knowing how many files are processed multiple times, since it just happens to be that a warning is generated here that's easy to see.

### PR DESCRIPTION
**Pull request by rondale-sc**
Should fix https://github.com/salsify/ember-css-modules/issues/63

When `loadPath` is called the `seen` object is meant to capture the
absolute path as a property and then check to see if that prop exists
before trying again.

Before this the `seen` object was not updated until the promises were
all resolved which meant that the guard to prevent multiple checks was
never hit

This fixes the problem  by setting the key on the `seen` object to a
promise that resolves the result of `load`.

/ht @mixonic for help with the promise solution

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -123,19 +123,20 @@ CSSModules.prototype.fetchExports = function(importPath, fromFile) {
...
+  var encoding = this.encoding;
+
   var absolutePath = dependency.toString();
-  if (seen[absolutePath]) {
-    return Promise.resolve(seen[absolutePath]);
+  var loadPromise = seen[absolutePath];
+
+  if (!loadPromise) {
+    loadPromise = new Promise(function(resolve) {
+      var content = fs.readFileSync(absolutePath, this.encoding);
```
Comment (mixonic) [inline index.js:131]: If `this.encoding` is used here, declaring it as a variable above is not required.

**Comment by rondale-sc** [comment]
Ah, quite right.  Surprised jshint didn't pick that up.

File: index.js (hunk 1 above)
Comment (rondale-sc) [reply index.js:131]: Good catch, I would have thought jshint would have seen that.

**Comment by dfreeman** [comment]
@rondale-sc 🤦‍♂️ I have almost this exact same fix on a local branch that I completely forgot about. Thanks so much to both you and @mixonic for taking care of this!

**Comment by dfreeman** [comment]
Published as `broccoli-css-modules@0.5.5` (don't ask what happened to 0.5.4... yarn and I got in a fight and that's all I'm saying 😉)

The next version of ember-css-modules that goes out will have the version constraint bumped, but for now a fresh install should pick up the fix. Thanks again!

**Comment by rondale-sc** [comment]
Haha, sounds great!  Thanks dfreeman!

**Pull request merged**